server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -512,10 +512,10 @@ private void addSequencesFromCache(
                     return Iterators.emptyIterator();
                   }
 
-                  return objectMapper.readValues(
+                  return ((List<Object>) objectMapper.readValue(
```

Comment:
I think `readValues` will still work here. Its docs say "Sequence can be either root-level 'unwrapped' sequence (without surrounding JSON array), or a sequence contained in a JSON Array."